[New Product] SQLite

adriens opened the pull request:
cf : 

- https://github.com/endoflife-date/endoflife.date/issues/2441

adriens committed:
[Request - Product] SQLite feather

marcwrobel commented on products/sqlite.md:
```diff
@@ -0,0 +1,333 @@
+
+> The code for SQLite is in the public domain and is thus free for use for any purpose, commercial or private
+
+
```
A paragraph explaining the release / support policy is needed.

adriens replied:
Done :zap:

marcwrobel commented on products/sqlite.md:
```diff
@@ -0,0 +1,333 @@
+    releaseDate: 2001-09-28
+
+-   releaseCycle: "1.0"
+    eol: false
```
Is 1.x version, with latest on 2001-07-23, still supported ?

The same comment applies to all cycles.

adriens replied:
According to their [Long Term Support ](https://www.sqlite.org/lts.html) it looks like they all are supported : do you share the same analysis ?

marcwrobel replied:
I do not have the same opinion about the LTS policies. I they are just saying they plan to develop SQLite as if the project will be supported until 2050.

You should be looking at https://sqlite.org/com/tech.html :

>  We will support any version of SQLite up to a year old, **not just the latest version**. At your request, we will backport any bug fixes to any SQLite version that is less than one year old. 

Given that information I also think that the cycles must be the `major` version, not `major.minor`. It will be easier to maintain.

marcwrobel replied:
Versions collapsed and description updated.

marcwrobel commented on products/sqlite.md:
```diff
@@ -0,0 +1,333 @@
+    latestReleaseDate: 2001-11-04
+    releaseDate: 2001-09-28
+
+-   releaseCycle: "1.0"
```
`1` would be best.

adriens replied:
Fixed.

marcwrobel commented on products/sqlite.md:
```diff
@@ -0,0 +1,333 @@
+    latestReleaseDate: 2004-10-12
+    releaseDate: 2004-06-18
+
+-   releaseCycle: "2.08"
```
Are multiple minor versions of the same major supported in parallel ? If no, `2` cycles should be collapsed.

adriens replied:
According to their [LTS Policies](https://www.sqlite.org/lts.html), I may say that they do : do you share the same analysis ?

marcwrobel replied:
I do not have the same opinion about the LTS policies. I they are just saying they plan to develop SQLite as if the project will be supported until 2050.

You should be looking at https://sqlite.org/com/tech.html :

>  We will support any version of SQLite up to a year old, **not just the latest version**. At your request, we will backport any bug fixes to any SQLite version that is less than one year old. 

Given that information I also think that the cycles must be the `major` version, not `major.minor`. It will be easier to maintain.

marcwrobel replied:
Versions collapsed and description updated.

marcwrobel commented on products/sqlite.md:
```diff
@@ -0,0 +1,333 @@
+  # https://rubular.com/r/KlemgnguNe0e5X
+  #  regex: ^REL_?(?<major>[1-9]\d*)_(?<minor>0|[1-9]\d*)_?(?<patch>\d+)?$
+releases:
+-   releaseCycle: "3.40"
```
Are multiple minor versions of the same major supported in parallel ? If no, `3` cycles should be collapsed.

adriens replied:
According to their [LTS Policies](https://www.sqlite.org/lts.html), I may say that they do : do you share the same analysis ?

marcwrobel replied:
See https://github.com/endoflife-date/endoflife.date/pull/2442#discussion_r1097150292

marcwrobel replied:
Versions collapsed and description updated.

marcwrobel commented on products/sqlite.md:
```diff
@@ -0,0 +1,333 @@
+
+> SQLite is an in-process library that implements a self-contained, serverless, zero-configuration, 
+> transactional SQL database engine.
+
```
This empty line should be removed (or it will be considered two distinct citations).

marcwrobel commented on products/sqlite.md:
```diff
@@ -0,0 +1,333 @@
+category: db
+iconSlug: sqlite
+permalink: /sqlite
+releasePolicyLink: https://www.sqlite.org/news.html
```
Is there a page on the SQLite explaining their release policy ? If yes this link should be used, if no using `https://www.sqlite.org/` would be best.

adriens replied:
Fixed :+1:

marcwrobel replied:
I still see the same link.

Also found this one which would be more appropriate : https://sqlite.org/com/tech.html.

marcwrobel replied:
Finally used https://www.sqlite.org/support.html.

marcwrobel commented on products/sqlite.md:
```diff
@@ -0,0 +1,333 @@
+iconSlug: sqlite
+permalink: /sqlite
+releasePolicyLink: https://www.sqlite.org/news.html
+changelogTemplate: https://sqlite.org/releaselog/__LATEST__/.html
```
This generate wrong links, such as https://sqlite.org/releaselog/2.00.08/.html.

adriens replied:
Yes, it's because I did add trailing `0` for better sorting... I remove them :zap:

adriens replied:
There is this page : https://www.sqlite.org/draft/chronology.html but some tas are missing too... :sweat:

adriens replied:
... and we cannot fin them on https://github.com/sqlite/sqlite/tags ...

marcwrobel replied:
It does not matter if some links are absent, just use `link: null` in that case.

marcwrobel replied:
Used https://www.sqlite.org/changes.html instead. It contains all the versions with anchors.

marcwrobel commented on products/sqlite.md:
```diff
@@ -0,0 +1,333 @@
+versionCommand: sqlite3 --version
+releaseDateColumn: true
+auto:
+-   git: https://github.com/sqlite/sqlite.git
```
Given the tag name (such as `version-3.40.1`) a regex is needed.

adriens replied:
All tags are not available on GH : what should be done ?

marcwrobel replied:
What is the range of available tags ?

adriens replied:
> Are multiple minor versions of the same major supported in parallel ? If no, 3 cycles should be collapsed.

I could first intent a merge in single majors versions @marcwrobel :grey_question:

marcwrobel replied:
Yes, that would be better to have only major version cycles IMHO.

marcwrobel replied:
Regex added.

marcwrobel reviewed (changes requested):
@adriens, thank you for this PR, I left you a few notes.

Remember there is a preview available on https://deploy-preview-2442--endoflife-date.netlify.app/sqlite.

adriens committed:
Details about LTS

cf https://github.com/endoflife-date/endoflife.date/pull/2442#discussion_r1095643933

adriens committed:
Blank line

cf https://github.com/endoflife-date/endoflife.date/pull/2442#discussion_r1095647779

adriens committed:
1.0 becomes 1

cf https://github.com/endoflife-date/endoflife.date/pull/2442#discussion_r1095644938

marcwrobel commented on products/sqlite.md:
```diff
@@ -0,0 +1,349 @@
+iconSlug: sqlite
+permalink: /sqlite
+alternate_urls:
+-   /sqlite3
```
Should be removed : [sqlite3](https://docs.python.org/3/library/sqlite3.html#module-sqlite3) is a python library.

marcwrobel replied:
I was wrong, see https://en.wikipedia.org/wiki/SQLite.

marcwrobel committed:
[sqlite] Update releasePolicyLink

https://sqlite.org/com/tech.html is the only page where it is stated that only the latest version is supported.

marcwrobel committed:
Remove sqlite3 alternate_url

sqlite3 is a python library.

marcwrobel committed:
Normalize content (#2124)

marcwrobel committed:
[sqlite] Collapse releases

marcwrobel committed:
[sqlite] Fix changelogtemplate

marcwrobel committed:
[sqlite] Add back sqlite3 alternate url

marcwrobel committed:
[sqlite] Rewrite description

marcwrobel committed:
Update releasePolicyLink to the main support page

marcwrobel reviewed (approved):
LGTM, but requires a review by @adriens and by another @endoflife-date/everyone collaborator.

usta commented on products/sqlite.md:
```diff
@@ -0,0 +1,53 @@
+---
+
+> SQLite is an in-process library that implements a self-contained, serverless, zero-configuration,
+> transactional SQL database engine. The code for SQLite is in the public domain and is thus free
```
` and is thus` looks like a typo here , could someone who is native english speaker can check for us ?

marcwrobel replied:
LGTM : https://www.merriam-webster.com/dictionary/thus.

usta commented on products/sqlite.md:
```diff
@@ -0,0 +1,53 @@
+    releaseDate: 2004-06-18
+
+-   releaseCycle: "2"
+    releaseDate: 2001-09-28
```
@marcwrobel isnt this page also need a bit  Normalization ?

marcwrobel replied:
Why ? It looks already normalized to me.

marcwrobel committed:
[sqlite] Normalization
